programs/drift/src/controller/orders.rs

```diff
@@ -3228,6 +3244,37 @@ pub fn fulfill_spot_order_with_match(
         return Ok(0_u64);
     }
 
+    let (taker_max_base_asset_amount, taker_max_quote_asset_amount) =
+        get_max_fill_amounts(taker, taker_order_index, base_market, quote_market)?;
+
+    let taker_implied_max_base_asset_amount =
+        if let Some(taker_max_quote_asset_amount) = taker_max_quote_asset_amount {
+            taker_max_quote_asset_amount
+                .safe_mul(base_market.get_precision())?
+                .safe_div(maker_price)?
+        } else {
+            u64::MAX
+        };
+    let taker_base_asset_amount = taker_base_asset_amount
+        .min(taker_max_base_asset_amount.unwrap_or(u64::MAX))
+        .min(taker_implied_max_base_asset_amount);
+
+    let (maker_max_base_asset_amount, maker_max_quote_asset_amount) =
+        get_max_fill_amounts(maker, maker_order_index, base_market, quote_market)?;
+
+    let maker_implied_max_base_asset_amount =
+        if let Some(maker_max_quote_asset_amount) = maker_max_quote_asset_amount {
+            maker_max_quote_asset_amount
+                .safe_mul(base_market.get_precision())?
+                .safe_div(maker_price)?
+        } else {
+            u64::MAX
```

Comment:
when would this get hit?

Reply:
the maker_max_quote_asset_amount == None when the maker is shorting base, in which case the `maker_max_base_asset_amount` should be Some